testsuite: attempt fix of statwatcher-based waitfile script

grondo opened the pull request:
The waitfile script held a vestige of a polling mode version that checked mtime of the target compared to the previous polling interval. With statwatcher, waitfile may wake up immediately on changes to the target file, and if this happens within one second of the previous check a false negative was returned from the `check()` function.

Remove the use of mtime, which shouldn't be required anyway if statwatcher is working.

This will likely resolve false timeouts as described in #569 -- however, I'd like to make sure we don't see any other failures for awhile before calling that Issue resolved.

I'd also like to attempt to write a testsuite for the waitfile script itself.

garlick commented:
Can we pull this in now, or did you want to write the test first?  (I'm fine with deferring the test)

grondo commented:
I haven't had time (and also I'm not really sure how) to write the tests. However, let me at least sanity check by hand then give an OK here.

grondo commented:
OK, I feel better about this now. I rebased the PR and added a couple sanity check tests for `waitfile.lua` to `t0001-basic.t`. This is ready to merge if travis is green.
